src/txParsers.ts

```diff
@@ -256,34 +278,73 @@ export const parseTxBody = (unparsedTxBody: unknown): TransactionBody => {
         metadataHash: parseOptional(unparsedTxBody.get(7), parseMetadataHash),
         validityIntervalStart: parseOptional(unparsedTxBody.get(8), parseValidityIntervalStart),
         mint: parseOptional(unparsedTxBody.get(9), parseMint),
+        scriptDataHash: parseOptional(unparsedTxBody.get(11), parseScriptDataHash),
+        collaterals: parseOptional(unparsedTxBody.get(13), parseCollaterals),
+        requiredSigners: parseOptional(unparsedTxBody.get(14), parseRequiredSigners),
+        networkId: parseOptional(unparsedTxBody.get(15), parseNetworkId),
     }
 }
 
 export const parseTx = (unparsedTx: unknown): Transaction => {
-    const [body, witnessSet, auxiliaryData] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedTx,
         ParseErrorReason.INVALID_TX_CBOR,
         parseTxBody,
         dontParse,
         dontParse,
+        dontParse,
     )
+    const presentItems = otherItems.filter((item) => item !== undefined)
 
-    return {body, witnessSet, auxiliaryData}
+    // cardano-cli with --shelley-era, --allegra-era and --mary-era
+    // includes only txBody, witnessSet and auxiliaryData
+    if (presentItems.length === 2) {
+        return { body, witnessSet: presentItems[0], auxiliaryData: presentItems[1] }
+    }
+
+    // cardano-cli with --alonzo-era includes all tx items
+    return {
+        body,
+        witnessSet: presentItems[0],
+        scriptValidity: presentItems[1],
+        auxiliaryData: presentItems[2],
+    }
 }
 
 export const parseRawTx = (unparsedRawTx: unknown): RawTransaction => {
-    const [body, item1, item2] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedRawTx,
         ParseErrorReason.INVALID_RAW_TX_CBOR,
         parseTxBody,
         dontParse,
         dontParse,
+        dontParse,
+        dontParse,
+        dontParse,
     )
-    // older versions of cardano-cli did not include scriptWitnesses
-    if (item2 === undefined) {
-        return {body, scriptWitnesses: undefined, auxiliaryData: item1}
+    const presentItems = otherItems.filter((item) => item !== undefined)
+
+    // older versions of cardano-cli included only txBody and auxiliaryData
+    if (presentItems.length === 1) {
+        return { body, auxiliaryData: presentItems[0] }
     }
+
     // cardano-cli expects indefinite-length scriptWitnesses
-    (item1 as any).encodeCBOR = Encoder.encodeIndefinite
-    return {body, scriptWitnesses: item1, auxiliaryData: item2}
+    (presentItems[0] as any).encodeCBOR = Encoder.encodeIndefinite
+
+    // newer versions of cardano-cli with --shelley-era, --allegra-era and --mary-era
+    // include txBody, scriptWitnesses and auxiliaryData
+    if (presentItems.length === 2) {
+        return { body, scriptWitnesses: presentItems[0], auxiliaryData: presentItems[1] }
+    }
+
+    // newer versions of cardano-cli with --alonzo-era include all rawTx items
+    return {
+        body,
+        scriptWitnesses: presentItems[0],
```

Comment:
Does this work if we use a mix of native scripts and Plutus scripts?

Reply:
The sad reality is that cardano-cli throws all nativeScriptWitnesses as well as plutusScriptWitnesses into this single rawTx item (that's also a reason why I renamed it from `nativeScriptWitnesses` to `scriptWitnesses`); and then partitions it when creating a signed transaction. The partitioning algorithm (i.e. how does it know which witnesses are of which type) will need further investigation.